mbuild/env.py

```diff
@@ -903,13 +893,14 @@ def update(self, targets=None):
                                   ap.group('value'), 'equals' ))
                 continue
             sp = env_t.supplement_pattern.match(t)
-            if ap:
+            if sp:
                 msgb("BINDING", "%s --> [%s]" % 
-                     (ap.group('name'), ap.group('value')))
-                bindings.append( (ap.group('name'), 
-                                  ap.group('value'), 'plusequals') )
+                     (sp.group('name'), sp.group('value')))
+                bindings.append( (sp.group('name'),
+                                  sp.group('value'), 'plusequals') )
```

Comment:
Supplement patterns were never checked, so I went ahead and fixed that. Let me know if that's not necessary.